evalml/automl/automl_algorithm/iterative_algorithm.py

```diff
@@ -69,10 +88,23 @@ def add_result(self, score_to_minimize, pipeline):
             score_to_minimize (float): The score obtained by this pipeline on the primary objective, converted so that lower values indicate better pipelines.
             pipeline (PipelineBase): The trained pipeline object which was used to compute the score.
         """
-        super().add_result(score_to_minimize, pipeline)
+        if pipeline.model_family != ModelFamily.ENSEMBLE:
+            super().add_result(score_to_minimize, pipeline)
         if self.batch_number == 1:
             self._first_batch_results.append((score_to_minimize, pipeline.__class__))
 
+        if pipeline.model_family not in self._best_pipeline_params and score_to_minimize is not None:
```

Comment:
Haha I think this came out of the `test_pipelines_in_batch_return_none` test which checks what if the pipeline returns None 😅  When None is returned, we get an `TypeError: '<' not supported between instances of 'NoneType' and 'int'`; np.nan doesn't throw this error since you can compare (though `score_to_minimize < current_best_score` will always be False so we never update the value to np.nan)